server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/ProcedureDivisionHeaderProcess.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.engine.processors;
+
+import org.eclipse.lsp.cobol.common.error.ErrorSeverity;
+import org.eclipse.lsp.cobol.common.error.ErrorSource;
+import org.eclipse.lsp.cobol.common.error.SyntaxError;
+import org.eclipse.lsp.cobol.common.message.MessageTemplate;
+import org.eclipse.lsp.cobol.common.model.NodeType;
+import org.eclipse.lsp.cobol.common.model.SectionType;
+import org.eclipse.lsp.cobol.common.model.tree.FunctionReference;
+import org.eclipse.lsp.cobol.common.model.tree.Node;
+import org.eclipse.lsp.cobol.common.model.tree.SectionNode;
+import org.eclipse.lsp.cobol.common.model.tree.variable.VariableUsageNode;
+import org.eclipse.lsp.cobol.common.processor.ProcessingContext;
+import org.eclipse.lsp.cobol.common.processor.Processor;
+
+/**
+ * Check semantics of variables used under {@link
+ * org.eclipse.lsp.cobol.common.model.tree.ProcedureDivisionUsingNode} and {@link
+ * org.eclipse.lsp.cobol.common.model.tree.ProcedureDivisionReturningNode}
+ */
+public class ProcedureDivisionHeaderProcess implements Processor<Node> {
```

Comment:
I suggest picking a better name for this class. This one does not seem to be related at all to what it does.
As far as I can tell, this class validates/checks the origin of variables used in the function/program calls. So maybe `LinkageArgumentsCheck` or `LinkageArgumentsOriginCheck`?

Reply:
Thanks! updated as per suggestion